libc/baselibc/src/tinyprintf.c

```diff
@@ -282,7 +282,10 @@ size_t tfp_format(FILE *putp, const char *fmt, va_list va)
             }
 
             /* Width */
-            if (ch >= '0' && ch <= '9') {
+            if (ch == '*') {
+                p.width = intarg(0, 1, &va);
```

Comment:
It might be a good idea to do a range check here.  A standard-compliant `printf` allows an `int` variable width, but tinyprintf's `width` member is an `unsigned char`.  I'm *not* suggesting you change this to support the full `int` range (that doesn't seem like a worthwhile change).  Perhaps this code could clamp the width to `[0, UCHAR_MAX]`.

Reply:
Yes, this is a good idea. I'll merge this one since it's already approved and address this in separate PR soon.